swap_meet/vendor.py

```diff
@@ -1,2 +1,51 @@
 class Vendor:
-    pass
+    def __init__(self, inventory=[]):
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        category_items = []
+
+        for item in self.inventory:
+            if item.category ==  category:
+                category_items.append(item)
+        return category_items
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+        else:
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+            vendor.inventory.remove(their_item)
+            vendor.inventory.append(my_item)
+            return True 
+    
+    def swap_first_item(self, vendor):
+        if len(self.inventory) == 0 or len(vendor.inventory) == 0:
+            return False
+        else:
+            self.swap_items(vendor, self.inventory[0], vendor.inventory[0])
```

Comment:
Nice function reuse!

This method is an interesting look at the balance between code re-use or custom implementations that take into account additional information for better performance. We _could_ choose to reuse our previous code. _Or_ we could specialize our implementation to achieve better performance for this particular scenario. Is there a O(1) implementation for this specific scenario?